src/main/java/org/folio/service/fieldname/FiledNamesServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package org.folio.service.fieldname;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.collections4.MapUtils;
+import org.folio.rest.jaxrs.model.FieldName;
+import org.folio.rest.jaxrs.model.FieldName.RecordType;
+import org.folio.rest.jaxrs.model.FieldNameCollection;
+import org.folio.service.fieldname.builder.DisplayNameKeyBuilder;
+import org.folio.service.fieldname.builder.FieldIdBuilder;
+import org.folio.service.fieldname.builder.PathBuilder;
+import org.folio.util.OkapiConnectionParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.folio.rest.jaxrs.model.FieldName.RecordType.HOLDINGS;
+import static org.folio.rest.jaxrs.model.FieldName.RecordType.INSTANCE;
+import static org.folio.rest.jaxrs.model.FieldName.RecordType.ITEM;
+import static org.folio.service.fieldname.FieldNameConfig.FIXED_LENGTH_DATA_ELEMENT;
+import static org.folio.service.fieldname.FieldNameConfig.HR_ID;
+import static org.folio.service.fieldname.FieldNameConfig.IDENTIFIERS;
+
+@Service
+public class FiledNamesServiceImpl implements FieldNamesService {
+
+  public static final String REFERENCE_DATA_NAME_KEY = "name";
+  private static final Set<FieldNameConfig> INSTANCE_FIELD_NAME_CONFIGS = EnumSet.of(
```

Comment:
We better move these constants to a separate file. If it's too much in this PR let's make to do it in the next PRs